tests/kafkatest/tests/mirror_maker_test.py

```diff
@@ -0,0 +1,143 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from ducktape.utils.util import wait_until
+from ducktape.mark import parametrize, matrix
+
+from kafkatest.services.zookeeper import ZookeeperService
+from kafkatest.services.kafka import KafkaService
+from kafkatest.services.console_consumer import ConsoleConsumer, is_int
+from kafkatest.services.verifiable_producer import VerifiableProducer
+from kafkatest.services.mirror_maker import MirrorMaker
+from kafkatest.tests.produce_consume_validate import ProduceConsumeValidateTest
+
+import time
+
+
+def bounce(test, clean_shutdown=True):
+    """Bounce mirror maker with a clean (kill -15) or hard (kill -9) shutdown"""
+
+    # Wait until messages start appearing in the target cluster
+    wait_until(lambda: len(test.consumer.messages_consumed[1]) > 0, timeout_sec=15)
+
+    # Wait for at least one offset to be committed
```

Comment:
Given how much time was spent discovering this issue, it might warrant a slightly longer comment. Specifically, don't just say what you're doing, also explain why. Even just a link to the JIRA would be helpful.